std/datetime/date.d

```diff
@@ -9858,23 +9858,17 @@ private int cmpTimeUnitsCTFE(string lhs, string rhs) @safe pure nothrow @nogc
     import std.format : format;
     import std.meta : AliasSeq;
 
-    static string genTest(size_t index)
+    static assert(timeStrings.length == 10);
+    static foreach (i; 0 .. timeStrings.length)
```

Comment:
Naming the indexes `curr`, `prev` and `next`, respectively would make the code a bit clearer.